doc: add more context for the witness feature limitation (#536)
harvester/docs: docs/advanced/witness.md

The general upgrade requirements and procedures apply to clusters with a witness node. However, the existence of degraded volumes in such clusters may cause upgrade operations to fail. 

> ## Longhorn Replicas with a Witness Node
> 
> The Longhorn storage system is a highly avaiable distributed storage system that create replicas for volumes. Longhorn is provisioned to management or worker nodes, but not to witness nodes. The default replica number is set to 3. That means in following scenarios, 3 nodes cluster with 1 witness node, Longhorn will not be able to create the default number of replicas. This will cause the potential risk for high availability. Please check on the following section to get more details.

Reviewer: ```suggestion
## Longhorn Replicas in Clusters with a Witness Node

Harvester uses Longhorn, a distributed block storage system, for management of block device volumes. Longhorn is provisioned to management and worker nodes but not to witness nodes, which do not store any data.

Longhorn creates replicas of each volume to increase availability. Replicas contain a chain of snapshots of the volume, with each snapshot storing the change from a previous snapshot. In Harvester, the default StorageClass `harvester-longhorn` has a replica count value of `3`.
```

## Limitations

> The default StorageClass `harvester-longhorn` of Harvester has a replica count value of `3` for high availability. According to above, if you use this StorageClass to create volumes with 2 nodes + 1 witness node, Longhorn is unable to create the configured number of replicas. This results in volumes being marked as **Degraded** on the Longhorn UI. If your cluster have more than 3 nodes, the volume will be created successfully and **healthy**.

Reviewer: ```suggestion
Witness nodes do not store any data. This means that in three-node clusters (no worker nodes), only two replicas are created for each Longhorn volume. However, the default StorageClass `harvester-longhorn` has a replica count value of `3` for high availability. If you use this StorageClass to create volumes, Longhorn is unable to create the configured number of replicas. This results in volumes being marked as **Degraded** on the Longhorn UI.
```

> The witness node does not store any data so only two replicas are created for each Longhorn volume in the cluster. However, the default StorageClass `harvester-longhorn` has a replica count value of `3` for high availability. If you use this StorageClass to create volumes, Longhorn is unable to create the configured number of replicas. This results in volumes being marked as **Degraded** on the Longhorn UI.
> 
> The above case only happens when you use the three-node cluster, which means this cluster does not contain any worker nodes. The Longhorn replica could be scheduled for either the worker or management nodes.

Reviewer: ```suggestion
The witness node does not store any data. This means that in three-node clusters (no worker nodes), only two replicas are created for each Longhorn volume. However, the default StorageClass `harvester-longhorn` has a replica count value of `3` for high availability. If you use this StorageClass to create volumes, Longhorn is unable to create the configured number of replicas. This results in volumes being marked as **Degraded** on the Longhorn UI.
```

Reviewer: It is better to add the new information to the existing paragraph.

> :::note
> It can be summarized as follows:
> - Three-node cluster (two management nodes and one witness node): You need to create a new StorageClass with the replica count set to `2`.
> - More than three nodes cluster: No need to do anything.
> :::
> 
> To resolve this issue, create a new *default* StorageClass with the **Number of Replicas** parameter set to **2**. This ensures that only two replicas are created for each Longhorn volume.

Reviewer: ```suggestion
In summary, you must use a StorageClass that matches the cluster configuration.

- 2 management nodes + 1 witness node: Create a new default StorageClass with the **Number of Replicas** parameter set to **2**. This ensures that only two replicas are created for each Longhorn volume.
- 2 management nodes + 1 witness node + 1 or more worker nodes: You can use the existing default StorageClass.
```
